### DISCUSSION TITLE
Having all possible options

### DISCUSSION THREAD

**Discussion by rmjstn** [keycloak/keycloak#12308] [Archive]
Hello,

The options `spi-user-profile-legacy-user-profile-read-only-attributes` and `spi-user-profile-legacy-user-profile-admin-read-only-attributes` are defined [here](https://www.keycloak.org/docs/latest/server_admin/#_read_only_user_attributes).

These options are not reported by the command `kc.sh start --help`.

Is there any way to get a complete list of these options ?

For example I'm looking for option to turn off the Keycloak theme cache in the Quarkus distribution (like     staticMaxAge, cacheThemes and cacheTemplates)

Thanks,

Rémy

**Comment by psytester** [comment]
👍 such convenience operation would help a lot ❤️
In wildfly it was possible to use the wildfly CLI to walk through the spi paths or the harder way looking into (standalone) xml file finding all present spi names and sub elements.

**Comment by mazenaissa** [comment]
Hello @rmjstn ,
Here are the options for theme caching
--spi-theme-static-max-age=-1 --spi-theme-cache-themes=false --spi-theme-cache-templates=false
found here
https://www.keycloak.org/docs/latest/server_development/#_themes
![image](https://user-images.githubusercontent.com/25006500/188195121-5535ea8e-4e43-4115-8170-5a2ef400aa81.png)

**Comment by lexcao** [comment]
Hi @rmjstn 
If you are finding a way to configure the SPI.
Here is the [doc](https://www.keycloak.org/server/configuration-provider) on configuration option format.
